### PR TITLE
Add Beam Privacy Wallet

### PR DESCRIPTION
**Pull request by Webonnix**

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.webonnix.beamprivacy

**Comment by barthalion** [comment]
The filename of manifest has to match `app-id` set inside.

File: org.webonnix.beamprivace.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+app-id: org.beamprivacy.desktop
+runtime: org.kde.Platform
+runtime-version: '3.1.5765'
```
Comment (barthalion) [inline org.webonnix.beamprivace.yaml]: I'm sure it's not what you want. It is for specifying runtime version, not application version.

File: org.webonnix.beamprivace.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+sdk: org.kde.Sdk
+command: beamprivacy 
+rename-icon: beamprivacy 
+rename-desktop-file: beamprivacy.desktop
```
Comment (barthalion) [inline org.webonnix.beamprivace.yaml]: These renames should happen upstream.

File: org.webonnix.beamprivace.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=x11
+  - --share=ipc
+  - --device=dri
+  - --filesystem=host # write output files
```
Comment (barthalion) [inline org.webonnix.beamprivace.yaml]: Why do you need so wide filesystem access?

File: org.webonnix.beamprivacy.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.webonnix.beamprivacywallet</id>
```
Comment (barthalion) [inline org.webonnix.beamprivacy.appdata.xml]: The id here needs to match manifest filename/app-id in the manifest.

File: org.webonnix.beamprivacy.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  <name>beamprivacywallet</name>
+  <summary>Beamp rivace wallet</summary>
```
Comment (barthalion) [inline org.webonnix.beamprivacy.appdata.xml]: Typo? Also summary could be improved.

File: org.webonnix.beamprivacy.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline org.webonnix.beamprivacy.appdata.xml]: Ideally it should be shipped upstream as other distributions could benefit from it.

File: org.webonnix.beamprivace.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  - --share=network
+
+modules:
+- name: beamprivacy  
+  buildsystem: qmake
+  sources:
+    - type: git
+      url: https://github.com/BeamMW/beam.git
+      branch: mainnet
```
Comment (barthalion) [inline org.webonnix.beamprivace.yaml]: As we want builds to be reproducible, you should use specific commit or tag here.

**Comment by LeandroStanger** [comment]
bot, build org.webonnix.beamprivacy

**Comment by LeandroStanger** [comment]
bot, build com.webonnix.beamprivacy

File: com.beamprivacy.desktop.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+app-id: com.beamprivacy.desktop
+runtime: org.gnome.Platform
+runtime-version: "3.32"
```
Comment (LeandroStanger) [inline com.beamprivacy.desktop.yaml]: Update runtime to 3.34

File: com.beamprivacy.desktop.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.webonnix.beamprivacywallet</id>
```
Comment (barthalion) [inline com.beamprivacy.desktop.appdata.xml]: ID here needs to match the one used in other places.

File: com.beamprivacy.desktop.yaml [hunk 10]
```diff
@@ -0,0 +1,19 @@
+app-id: com.beamprivacy.desktop
```
Comment (barthalion) [inline com.beamprivacy.desktop.yaml]: Is this an actual domain? Project seems to use https://beam.mw/ so it should be `mw.beam.Wallet` – ideally you should actually ask developers.

File: com.beamprivacy.desktop.yaml [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+command: beamprivacy 
+
+finish-args:
+  - --socket=x11
+  - --share=network
+
+modules:
+     sources:
+        - type: file
+          url: https://builds.beam.mw/mainnet/2019.08.08/Release/linux/Beam-Wallet-3.1.5765.deb
```
Comment (barthalion) [inline com.beamprivacy.desktop.yaml]: Unless it's difficult to achieve, building from source is preferred.

**Comment by barthalion** [comment]
Have you contacted upstream team about your submission?

**Pull request closed without merging**